packages/opentelemetry-exporter-collector/src/transform.ts

```diff
@@ -256,6 +328,209 @@ function toCollectorInstrumentationLibrarySpans(
+    labels: toCollectorLabels(metric.labels),
+    value: pointValue,
+    startTimeUnixNano: startTime,
+    timeUnixNano: core.hrTimeToNanoseconds(
+      metric.aggregator.toPoint().timestamp
+    ),
+  };
+}
+
+/**
+ * Returns a HistogramPoint to the collector
+ * @param metric
+ * @param startTime
+ */
+export function toHistogramPoint(
+  metric: MetricRecord,
+  startTime: number
+): opentelemetryProto.metrics.v1.HistogramDataPoint {
+  const histValue = metric.aggregator.toPoint().value as Histogram;
+  return {
+    labels: toCollectorLabels(metric.labels),
+    sum: histValue.sum,
+    count: histValue.count,
+    startTimeUnixNano: startTime,
+    timeUnixNano: core.hrTimeToNanoseconds(
+      metric.aggregator.toPoint().timestamp
+    ),
+    buckets: histValue.buckets.counts.map(count => {
+      return { count };
+    }),
+    explicitBounds: histValue.buckets.boundaries,
+  };
+}
+
+/**
+ * Converts a metric to be compatible with the collector
+ * @param metric
+ * @param startTime start time in nanoseconds
+ */
+export function toCollectorMetric(
+  metric: MetricRecord,
+  startTime: number
+): opentelemetryProto.metrics.v1.Metric {
+  // All data points are set as empty by default
+  let int64DataPoints: opentelemetryProto.metrics.v1.Int64DataPoint[] = [];
+  let doubleDataPoints: opentelemetryProto.metrics.v1.DoubleDataPoint[] = [];
+  let histogramDataPoints: opentelemetryProto.metrics.v1.HistogramDataPoint[] = [];
+  const descriptor = toCollectorMetricDescriptor(metric);
+
+  if (
+    descriptor.type ===
+    opentelemetryProto.metrics.v1.MetricDescriptorType.HISTOGRAM
+  ) {
+    histogramDataPoints = [toHistogramPoint(metric, startTime)];
+  } else if (metric.descriptor.valueType == apiValueType.INT) {
+    int64DataPoints = [toSingularPoint(metric, startTime)];
+  } else if (metric.descriptor.valueType === apiValueType.DOUBLE) {
+    doubleDataPoints = [toSingularPoint(metric, startTime)];
+  } // TODO: Add support for summary points once implemented
+
+  return {
+    metricDescriptor: toCollectorMetricDescriptor(metric),
+    doubleDataPoints,
+    int64DataPoints,
+    summaryDataPoints: [], // TODO: Add support for summary points once implemented
+    histogramDataPoints,
+  };
+}
+
+/**
+ * Prepares metric service request to be sent to collector
+ * @param metrics metrics
+ * @param startTime start time of the metric in nanoseconds
+ * @param collectorMetricExporterBase
+ */
+export function toCollectorExportMetricServiceRequest<
+  T extends CollectorExporterConfigBase
+>(
+  metrics: MetricRecord[],
+  startTime: number,
+  collectorMetricExporterBase: CollectorMetricExporterBase<T>
+): opentelemetryProto.collector.metrics.v1.ExportMetricsServiceRequest {
+  const groupedMetrics: Map<
+    Resource,
+    Map<core.InstrumentationLibrary, MetricRecord[]>
+  > = groupMetricsByResourceAndLibrary(metrics);
+  const additionalAttributes = Object.assign(
+    {},
+    collectorMetricExporterBase.attributes || {},
```

Comment:
Think there is no need for `|| {}` here. If null or undefined, it will have the same behavior I believe?